src/js/db/DBD.js

```diff
@@ -0,0 +1,170 @@
+const util = require('util');
+const log = require('../log');
+const core = require('../core');
+const generics = require('../generics');
+
+module.exports = {
+	/**
+	 * Generates a DB schema based upon WoWDBDefs.
+	 * @param {string} tableName
+	 * @param {string} layoutHash
+	 * @returns {object, object}
+	 */
+	getSchemaForTable: async (tableName, layouthash) => {
+		// Download DBD for specific table (keep case sensitivity in mind)
+		const url = util.format(core.view.config.dbdURL, tableName)
+		let data = await generics.downloadFile(url);
+
+		const targetBuild = core.view.casc.getBuildName();
+
+		// Parse DBD
+		const columnDefinitions = new Map();
+		const versionDefinitions = [];
+
+		const validTypes = ["int", "float", "string", "locstring"];
```

Comment:
Move this out of the function context, define it below the imports instead. No need to re allocate this array every time `getSchemaForTable` is called.